src/crypto/crypto_tls.cc

```diff
@@ -132,8 +135,11 @@ void KeylogCallback(const SSL* s, const char* line) {
 int NewSessionCallback(SSL* s, SSL_SESSION* sess) {
   TLSWrap* w = static_cast<TLSWrap*>(SSL_get_app_data(s));
   Environment* env = w->env();
-  HandleScope handle_scope(env->isolate());
-  Context::Scope context_scope(env->context());
+  Isolate* isolate = env->isolate();
+  HandleScope handle_scope(isolate);
+  Local<Context> context = env->context();
+  Context::Scope context_scope(context);
+  NoArrayBufferZeroFillScope no_zero_fill_scope(env->isolate_data());
```

Comment:
Putting this line here is a security vulnerability, because it affects the JS callback as well. `NoArrayBufferZeroFillScope` needs to be scoped as narrowly as possible.

Reply:
Thanks for the info!